### PR TITLE
Implementation/69 create server and client binaries 1/2

### LINKED ISSUE #69: Implementation - Create server and client binaries
- [x] Create two functional binaries `myteams_server` and `myteams_cli`
- [ ] Import `libnet_utils` with the server an run it
- [ ] Adapt `libnet_utils` for client side (add send to server etc... etc...)

## Low Priority
- [ ] #70

### LINKED ISSUE #82: Implemenation - Add codecov support
- Recover all coverage files
- Generate report with gcovr
- Upload generated report to codecov

### PR DESCRIPTION
**Pull request by Cavonstavant**
This first PR (one of two)  adds a fully functionnal network server with client addition / deletion both in the network layer and the core layer

Close #69 
Close #82

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,12 +20,12 @@ option(CODE_COVERAGE "Enable code coverage" OFF)
 if (CODE_COVERAGE)
     add_compile_options("--coverage" "-fprofile-arcs" "-ftest-coverage")
     add_link_options("--coverage" "-fprofile-arcs" "-ftest-coverage")
+    add_subdirectory(tests)
 endif()
 
 add_subdirectory(objects)
 add_subdirectory(client)
 add_subdirectory(server)
-add_subdirectory(tests)
```
Comment (LucasTesnier) [inline CMakeLists.txt:28]: Why removing the tests dir ?

File: server/net_utils/tcp_server.c [hunk 2]
```diff
@@ -19,10 +19,15 @@ static void bind_and_listen_on_socket(tcp_server_t *srv, long port)
...
+        return false;
+    }
+    if (listen(srv->sock_fd, LISTEN_BACKLOG) < 0) {
         TEAMS_LOG("listen");
+        return false;
+    }
+    return true;
 }
 
 bool add_user_to_server(tcp_server_t *srv, char *username, char *password)
```
Comment (LucasTesnier) [inline server/net_utils/tcp_server.c:33]: Add user to server ? Function seems to be server init no ?

File: server/src/my_teams_server.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+void teams_server_run(teams_server_t *self)
+{
+    tcp_server_t *network_srv = self->network_server;
+
+    server_fill_fd_sets(network_srv);
+    while (self->state) {
+        server_wait(network_srv);
+        if (server_manage_fd_update(network_srv))
+            self->add_user(self, fetch_last_added_peer(self->network_server));
+        server_fill_fd_sets(network_srv);
```
Comment (LucasTesnier) [inline server/src/my_teams_server.c:23]: Maybe Fill after remove the unused client no ?

File: server/src/my_teams_server.c [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    CIRCLEQ_FOREACH(client, &self->clients, clients) {
+        tmp = client;
+        client = CIRCLEQ_NEXT(client, clients);
+        CIRCLEQ_REMOVE(&self->clients, tmp, clients);
+        free(tmp->network_client);
+        free(tmp);
+    }
+}
+
+static void init_teams_server_methods(teams_server_t *new_server)
```
Comment (LucasTesnier) [inline server/src/my_teams_server.c:44]: Missing documentation for static ;)

File: server/src/my_teams_server.c [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+{
+    new_server->run = teams_server_run;
+    new_server->stop = teams_server_stop;
+    new_server->add_user = teams_server_add_user;
+}
+
+teams_server_t *create_new_server(long port, char *save_pathname)
+{
+    teams_server_t *new_server = malloc(sizeof(teams_server_t));
+    (void) save_pathname;
```
Comment (LucasTesnier) [inline server/src/my_teams_server.c:54]: Forgot to be removed

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some things need to be changes, otherwise good work man

File: CMakeLists.txt (hunk 1 above)
Comment (Cavonstavant) [reply CMakeLists.txt:28]: because

File: server/net_utils/tcp_server.c (hunk 2 above)
Comment (Cavonstavant) [reply server/net_utils/tcp_server.c:33]: This function will not be used since it's proper to the net_utils and not available in outer scope

Comment (LucasTesnier) [reply server/net_utils/tcp_server.c:33]: My bad, I don't know how to read a file correctly...

File: server/src/client_wrapper.c [hunk 6]
```diff
@@ -0,0 +1,40 @@
+/*
+** EPITECH PROJECT, 2022
+** mes_equipes
+** File description:
+** client_wrapper
+*/
+
+#include "client_wrapper.h"
+#include "my_teams_server.h"
+
```
Comment (osvegn) [inline server/src/client_wrapper.c:10]: \file

File: server/net_utils/tcp_server.c [hunk 7]
```diff
@@ -9,7 +9,7 @@
```
Comment (osvegn) [inline server/net_utils/tcp_server.c:9]: \file

File: server/src/entrypoint.c [hunk 8]
```diff
@@ -5,10 +5,29 @@
 ** entrypoint
 */
```
Comment (osvegn) [inline server/src/entrypoint.c:7]: \file

**Review by osvegn — CHANGES_REQUESTED** [review]
Some changes needed, but Look Good To Me

File: client/client_net_utils/client_utils.c [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+*/
+
+/// \file client/client_net_utils/client_utils.h
+
+#include "client_utils.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/// \brief juste as the function name says
```
Comment (osvegn) [inline client/client_net_utils/client_utils.c]: Need better documentation

**Review by osvegn — APPROVED** [review]
LGTM

**Review by LucasTesnier — APPROVED** [review]

**Pull request merged**